index.bs

```diff
@@ -175,6 +175,173 @@ partial interface Navigator {
+1. If any |element| of |audioSession|.[=AudioSession/[[elements]]=] has a [=default type=] of {{AudioSessionType/playback}} and its [=element state|state=] is {{AudioSessionState/active}}, return {{AudioSessionType/playback}}.
+1. If any |element| of |audioSession|.[=AudioSession/[[elements]]=] has a [=default type=] of {{AudioSessionType/transient}} and its [=element state|state=] is {{AudioSessionState/active}}, return {{AudioSessionType/transient}}.
+1. If any |element| of |audioSession|.[=AudioSession/[[elements]]=] has a [=default type=] of {{AudioSessionType/ambient}} and its [=element state|state=] is {{AudioSessionState/active}}, return {{AudioSessionType/ambient}}.
+1. Return {{AudioSessionType/auto}}.
+
+# Audio source and sink integration # {#integration}
+
+An <dfn>element state</dfn> is:
+* {{AudioSessionState/interrupted}} if it is in its {{AudioSession}}'s [=AudioSession/[[interruptedElements]]=].
+* {{AudioSessionState/active}} if it is an [=audible element=].
+* {{AudioSessionState/inactive}} otherwise.
+
+To <dfn>update an element</dfn> named |element|, the user agent MUST run the following steps:
+1. Let |audioSession| be |element|'s {{AudioSession}}.
+1. Run |element|'s [=element update steps|update steps=].
+1. If |element| is an [=audible element=] and |audioSession|.[=AudioSession/[[state]]=] is {{AudioSessionState/interrupted}}, run the following steps:
+    1. Add |element| to |audioSession|.[=AudioSession/[[interruptedElements]]=].
+    1. Run |element|'s [=element suspend steps|suspend steps=].
+1. If |element| is in |audioSession|.[=AudioSession/[[interruptedElements]]=], and |audioSession|.[=AudioSession/[[state]]=] is {{AudioSessionState/active}}, run the following steps:
+    1. Remove |element| from |audioSession|.[=AudioSession/[[interruptedElements]]=].
+    1. Run |element|'s [=element resume steps|resume steps=].
+
+When the [=audible flag=] of one of |audioSession|'s [=audio session/elements=] named |changingElement| is changing, the user agent MUST run the following steps:
+1. If |changingElement|'s [=audible flag=] is changing to `true`, [=try activating=] |audioSession| and abort these steps.
+1. Let |shouldAbort| be true if any |element| of |audioSession|.[=AudioSession/[[elements]]=] other than |changingElement| matches any of the following cases:
+    1. |element|'s [=element state|state=] is {{AudioSessionState/interrupted}}.
+    1. |element|'s [=audible flag=] is `true`.
+1. If |shouldAbort| is `true`, abort these steps.
+1. [=Inactivate=] |audioSession|.
+
+## AudioContext ## {#audiocontext-sink}
+
+An {{AudioContext}} is an [=audio session/element=] with the following properties:
+* Its [=default type=] is {{AudioSessionType/ambient}}.
+* Its [=audible flag=] is `true` if its state is {{AudioContextState/running}} and is sending non zero samples to its destination.
```

Comment:
We should add a note in [audible flag](https://pr-preview.s3.amazonaws.com/youennf/audio-session/pull/34.html#audible-flag) to mention that we will define the audible condition for each different elements later.

Reply:
I added a note in audio session element in [audio-session-element](https://pr-preview.s3.amazonaws.com/youennf/audio-session/pull/34.html#audio-session-element) for that.